File: VSMatch/Data/Entities/Feedback.cs
namespace VSMatch.Data.Entities;

public class Feedback
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Message { get; set; } = string.Empty;
    public FeedbackStatus Status { get; set; } = FeedbackStatus.New;
    public string? Reply { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public User? User { get; set; }
}

public enum FeedbackStatus
{
    New = 0,
    InProgress = 1,
    Resolved = 2,
}

File: frontend/src/api/feedback.ts
import { authFetch, throwApiError } from './client';

export type FeedbackStatus = 'New' | 'InProgress' | 'Resolved';

export interface FeedbackItem {
  id: string;
  userId: string;
  authorName: string;
  authorVkUserId: string | null;
  message: string;
  status: FeedbackStatus;
  reply: string | null;
  createdAt: string;
  updatedAt: string | null;
}

export async function createFeedback(message: string): Promise<FeedbackItem> {
  const res = await authFetch('/api/feedback', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ message }),
  });
  if (!res.ok) await throwApiError(res, 'Не удалось отправить обращение');
  return res.json();
}

export async function fetchAllFeedback(): Promise<FeedbackItem[]> {
  const res = await authFetch('/api/admin/feedback?pageSize=100');
  if (!res.ok) await throwApiError(res, 'Не удалось загрузить обращения');
  return res.json();
}

export async function updateFeedback(id: string, status: FeedbackStatus, reply: string | null): Promise<FeedbackItem> {
  const res = await authFetch(`/api/admin/feedback/${id}`, {
    method: 'PATCH',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ status, reply }),
  });
  if (!res.ok) await throwApiError(res, 'Не удалось обновить обращение');
  return res.json();
}
